src/lib.rs

```diff
@@ -798,9 +812,14 @@ impl<T: Counter> Histogram<T> {
             {
                 let c = self.mut_at(value).expect("value should fit after resize");
                 if add {
-                    *c = *c + count;
+                    // after resize, should be no possibility of overflow because this is a new slot
+                    *c = (*c).checked_add(&count).expect("count overflow after resize");
                 } else {
-                    *c = *c - count;
+                    // TODO Not sure this code path can ever be hit: if subtraction requires minuend
+                    // count to exceed subtrahend count for a given value, we shouldn't ever need
+                    // to resize to subtract.
+                    // Anyway, at the very least, we know it shouldn't underflow.
+                    *c = (*c).checked_sub(&count).expect("count underflow after resize");
```

Comment:
It would take more subtraction testing to make me feel confident that this is in fact unreachable, so I'd prefer to wait for #36 for that change